Piwik.Tracker/PiwikTracker.cs

```diff
@@ -1160,14 +1159,15 @@ public void SetIp(string ip)
         /// <exception cref="Exception"/>
         public void SetUserId(string userId)
         {
+            // todo this is contradicion: (userId == null) / string.IsNullOrEmpty(userId)

```

Comment:
What about changing it to this:
`public void SetUserId(string userId)
  {
            if (string.IsNullOrEmpty(userId))
            {
                throw new ArgumentException("User ID cannot be empty.", nameof(userId));
            }
            _userId = userId;
  }`
(removing the userId==null) case and letting the user use the existing method `SetNewVisitorId()` for this case?
Or do you think letting it as it is but with default parameter is better?